modular_skyrat/master_files/code/modules/pain_shock/status_indicator.dm

```diff
@@ -0,0 +1,198 @@
+
+/mob/living/proc/add_status_indicator(image/thing)
+	if(get_status_indicator(thing)) // No duplicates, please.
+		return
+	if(!istype(thing, /image))
+		thing = image(icon = 'modular_skyrat/master_files/icons/mob/status_indicators.dmi', loc = src, icon_state = thing)
+
+	LAZYADD(status_indicators, thing)
+	handle_status_indicators()
+
+// Similar to above but removes it instead, and nulls the list if it becomes empty as a result.
+/mob/living/proc/remove_status_indicator(image/thing)
+	thing = get_status_indicator(thing)
+
+	cut_overlay(thing)
+	LAZYREMOVE(status_indicators, thing)
+	handle_status_indicators()
+
+/mob/living/proc/get_status_indicator(image/thing)
+	if(!istype(thing, /image))
+		for(var/image/I in status_indicators)
+			if(I.icon_state == thing)
+				return I
+	return LAZYACCESS(status_indicators, LAZYFIND(status_indicators, thing))
+
+/mob/living/proc/cut_indicators_overlays()
+	if(!status_indicators) // sometimes the overlay cutting misses, so if theres nothing when its called, lets just clear them all!
+		var/list/static/potential_indicator = list(
+			"weakened",
+			"paralysis",
+			"stunnded",
+			"sleeping",
+			"confused",
+			"blinded",
+			"critical")
+		for(potential_indicator in overlays)
+			cut_overlay(potential_indicator)
+
+	else
+		for(var/thing in status_indicators)
+			cut_overlay(thing)
+
+// Refreshes the indicators over a mob's head. Should only be called when adding or removing a status indicator with the above procs,
+// or when the mob changes size visually for some reason.
+/mob/living/proc/handle_status_indicators()
+	// First, get rid of all the overlays.
+	cut_indicators_overlays()
+
+	if(!LAZYLEN(status_indicators))
+		return
+
+	if(stat == DEAD)
+		return
+	var/mob/living/carbon/carbon = src
+	var/icon_scale = get_icon_scale(carbon)
+	// Now put them back on in the right spot.
+	var/our_sprite_x = 16 * icon_scale
+	var/our_sprite_y = 24 * icon_scale
+
+	var/x_offset = our_sprite_x // Add your own offset here later if you want.
+	var/y_offset = our_sprite_y + STATUS_INDICATOR_Y_OFFSET
+
+	// Calculates how 'long' the row of indicators and the margin between them should be.
+	// The goal is to have the center of that row be horizontally aligned with the sprite's center.
+	var/expected_status_indicator_length = (STATUS_INDICATOR_ICON_X_SIZE * status_indicators.len) + (STATUS_INDICATOR_ICON_MARGIN * max(status_indicators.len - 1, 0))
+	var/current_x_position = (x_offset / 2) - (expected_status_indicator_length / 2)
+
+	// In /mob/living's `update_transform()`, the sprite is horizontally shifted when scaled up, so that the center of the sprite doesn't move to the right.
+	// Because of that, this adjustment needs to happen with the future indicator row as well, or it will look bad.
+	current_x_position -= (32 / 2) * (icon_scale - 1)
```

Comment:
Where does that `32 / 2` come from? Should likely be a define, because that's not going to change between the times the proc is ran.

Reply:
Gone next commit!